### ISSUE TITLE
Error while registering a Provider in TYPO3 7.4, Flux 7.2.1

### ISSUE THREAD

**Issue by boettner-it** [FluidTYPO3/flux#928]
```php
\FluidTYPO3\Flux\Core::registerProviderExtensionKey('Bit.' . $_EXTKEY, 'Page');
\FluidTYPO3\Flux\Core::registerProviderExtensionKey('Bit.' . $_EXTKEY, 'Content');
```

I get this error:
```
Fatal error: Call to a member function writeUC() on null in /usr/www/users/medenv/typo3/typo3_src-7.4.0/typo3/sysext/fluid/Classes/Core/Widget/AjaxWidgetContextHolder.php on line 62
```
Should I better avoid 7.4 yet? The error also occcurs on the current Flux development branch (7.2.2-beta)

**Comment by NamelessCoder** [comment]
You probably should write the vendor and extension name as hardcoded. The current registration is wrong because it results in something like `Bit.foobar` instead of `Bit.Foobar`.

And make sure you've got the most recent fluidcontent and flux which prevents access to the user session when it is not initialised. It won't fix your problem, but it will remove the fatal error.

**Comment by boettner-it** [comment]
Thanks for replying that quick.

> If your $_EXTKEY contains a _ it's bound to break.

that means having a ```_``` in the extkey is not possible at all?

The error persists with flux 7.2.2-beta and fluidcontent 4.3.2-beta, both development from GitHub and I hardcoded the extkey ```foo_bar``` to ```Bit.FooBar```

**Comment by NamelessCoder** [comment]
Extension *key* may contain underscores but the *extension name, with or without Vendor prefix, must always be in UpperCamelCase*. Make sure you clear all caches and temporary files and perform DB schema upgrade if necessary.

The error may not be related to FluidTYPO3 - it's being thrown by a standard Fluid widget component and FluidTYPO3 doesn't contain any widgets anymore.

**Comment by boettner-it** [comment]
Yes, I manually cleared temp files and caches as usual but:

> And make sure you've got the most recent fluidcontent and flux which prevents access to the user session when it is not initialised. It won't fix your problem, but it will remove the fatal error.

Are there a more recent version than fluidcontent 4.3.2 development and flux 7.2.2 development rom GitHub?

**Comment by NamelessCoder** [comment]
> Are there more recent versions than fluidcontent 4.3.2 development and flux 7.2.2 development from GitHub?

No, those are the most recent. If you upgraded the site you may need to flush `be_sessions` and `uc` from `be_users`, I've had that problem before (though without fatal errors).

If that doesn't work I'm afraid I have to refer you to the TYPO3 core bug tracker. The error comes while trying to write to the user configuration and is triggered from a Widget - FluidTYPO3 doesn't write to the user configuration, nor does it use any widgets. It could be caused by an incompatible Widget base class (of which FluidTYPO3 contains zero) in which dependency injection is broken (can't say without the stack trace) or use of PHP7 (which I've seen fail to perform dependency injection). Or maybe even a general problem related to use of Widgets with AJAX in a `USER_INT` object.

As a final tip, if you're using any Widgets in your own template setups, try using them in Partials which won't get parsed if the main template gets parsed. Generally speaking you shouldn't use Widgets in fluid content or page templates and if you do, it's advisable to separate the Widgets to Partials anyway simply for performance reasons (use of Widget means uncompilable template).

PS: No other users have reported this problem from an estimated ~100 users currently using the same versions as you use. Usually we hear about general fatal errors immediately so I'm pretty sure this is caused by some specific aspect of your site, templates, configuration, etc.

**Comment by boettner-it** [comment]
Yes, the cause for the error was a Widget and moving it into a partial made it work like before in TYPO3 6.2. 

Have a nice weekend!

**Comment by TrueType** [comment]
@boettner-it THX for the hint regardings partials. I just had the same/similar problem. I wanted to load provider templates in ext_tables.php with
```
if (FALSE === (boolean) $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['extprovider_tpl']['setup']['disableContentTemplates']) {
    \FluidTYPO3\Flux\Core::registerProviderExtensionKey('extprovider_tpl', 'Content');
}
```
Problem came with typo/init.php initilizes the BE_USER only after the ext_tables.php are processed. The templates in typo3conf/ext/extprovider_tpl/Resource/Private/Templates/Content/* are parsed and view helpers in the templates may use AjaxWidgetContextHolder, which needs a BE_USER. As partials are parsed later only, the error vanished by moving code from the template to a partial.

No idea if this is an official solution. Or if the loading within ext_tables.php should be done differently. Thanx anyway.

**Comment by boettner-it** [comment]
You are welcome but I just followed the suggestions @NamelessCoder gave me. :-)

**Comment by RozbehSharahi** [comment]
Same thing occurs on TYPO3 8.7.1 and FLUX 8.2.1.

Kind regards
Rozbeh Chiryai Sharahi

**Comment by biberesser** [comment]
Still happening :(
Typo3 9.5.14
Flux 9.3.2 / master
